### ISSUE TITLE
FatalErrorException with Laravel v5.2.23

### ISSUE THREAD

**Issue by timothyrowan** [formers/former#514]
Method Former\Form\Fields\Input::__toString() must not throw an exception, caught ErrorException: array_key_exists() expects parameter 2 to be array, null given

Same for any form or input type.

**Comment by claar** [comment]
Is this new behavior with this version of Laravel? Can you paste your Former code?

**Comment by timothyrowan** [comment]
I assume so, thats the only thing that has changed and all of a sudden started throwing errors.

    {!! Former::open_vertical()->action(route('auth.login'))->data_parsley_validate() !!}
    {!! Former::setOption('automatic_label', false) !!}
    <div class="form-group has-feedback">
        {!! Former::email('email')->placeholder('Email Address')->required()->data_parsley_type('email') !!}
        <span class="glyphicon glyphicon-envelope form-control-feedback"></span>
    </div>
    <div class="form-group has-feedback">
        {!! Former::password('password')->placeholder('Password')->required() !!}
        <span class="glyphicon glyphicon-lock form-control-feedback"></span>
    </div>
    <div class="row">
        <div class="col-xs-6">
            {!! Former::checkbox('remember')->class('switch')->inline() !!}
        </div>
        <div class="col-xs-6">
            {!! Former::button('Sign In')->type('submit')->class('btn btn-default btn-block btn-flat') !!}
        </div>
    </div>
    {!! Former::close() !!}

**Comment by claar** [comment]
Yep, I am able to reproduce this on a fresh install -- thanks for opening the ticket. Works on 5.2.22 but broke on 5.2.23.

For a workaround for now I'd suggest editing your `composer.json` to force Laravel 5.2.22.

**Comment by timothyrowan** [comment]
Awesome thanks for being so quick about this. Dunno how I'd live without former.

**Comment by vlakoff** [comment]
Could you please provide a code for reproducing?

**Comment by claar** [comment]
@vlakoff To reproduce, add `{!! Former::open() !!}` to a blade file after installing Former. For a more descriptive error message, use `{!! Former::open()->wrapAndRender() !!}`

**Comment by vlakoff** [comment]
Still can't reproduce, your code doesn't throw error. Otherwise, a stack trace would be very helpful.

**Comment by claar** [comment]
@vlakoff Oops, sorry, try:

```
    {!! Former::open()->wrapAndRender() !!}
    {!! Former::text('foo')->wrapAndRender() !!}
    {!! Former::close()->wrapAndRender() !!}
```

Relevant part of the stack trace is:

```
in Arr.php line 145
at HandleExceptions->handleError('2', 'array_key_exists() expects parameter 2 to be array, null given', '/home/vagrant/code/bakery/vendor/laravel/framework/src/Illuminate/Support/Arr.php', '145', array('array' => null, 'key' => 'inline'))
at array_key_exists('inline', null) in Arr.php line 145
at Arr::exists(null, 'inline') in Arr.php line 269
at Arr::get(null, 'inline', null) in helpers.php line 171
at array_get(null, 'inline') in Group.php line 430
at Group->getHelp() in Group.php line 178
at Group->wrapField(object(Input)) in Field.php line 159
at Field->wrapAndRender() in 80d5896250567bcb417a29f7129af5db8eea07f4.php line 44
```

**Comment by vlakoff** [comment]
Hitting another unrelated error, "session store not set on request", again.

Thanks for the stack, I'll have a look using it.

**Comment by vlakoff** [comment]
Yeah, looks like it's this `protected $help = null;` in Group.php.

**Comment by vlakoff** [comment]
You could either:
* replace protected `$help = null;` with `protected $help = array();`
* or add `if (!$this->help) { return ''; }` at the beginning of method `getHelp()`.

**Comment by vlakoff** [comment]
I'd vote for doing *both*.

**Comment by claar** [comment]
@vlakoff It's not hard to fix these 31 instances of using `array_get` that are now broken; that doesn't change the fact that Laravel broke backwards compatibility in a patch release (5.2.x).

Though I agree that passing nulls to `Arr::get` is sloppy, Laravel shouldn't be breaking backwards compatibility in a patch release.

**Comment by claar** [comment]
@vlakoff Not to mention that adding an `if (is_null($array)) return $default` to `Arr::get()` would both maintain backwards compatibility and make the `Arr::get()` interface friendlier.

**Comment by vlakoff** [comment]
* Not all the "31 instances" are "now broken": almost all receive only arrays.
* Your code was working by chance, you still should fix it how I described above.

Nevertheless, I'll think about adding a test for BC in 5.2 branch. But it should go with 5.3.

**Comment by claar** [comment]
Thanks, @vlakoff

**Comment by vlakoff** [comment]
PR https://github.com/laravel/framework/pull/12726.

But please fix your code as well!

**Comment by Renfos** [comment]
Same issue here.

**Comment by claar** [comment]
Fixed by @vlakoff in Former version 4.0.2. @vlakoff also fixed Laravel's patch release backwards incompatibility bug. You're awesome, @vlakoff, thank you.
